mitosheet/src/mito/types.tsx

```diff
@@ -184,11 +184,11 @@ export interface ColumnFilterMap {
 
 
 export enum NumberColumnFormatEnum {
-    PLAIN_TEXT = 'plain text',
-    CURRENCY = 'currency',
-    ACCOUNTING = 'accounting',
-    PERCENTAGE = 'percentage',
-    SCIENTIFIC_NOTATION = 'scientific notation'
+    PLAIN_TEXT = 'FORMAT_STRING_PLACEHOLDER',
+    CURRENCY = '$FORMAT_STRING_PLACEHOLDER',
+    ACCOUNTING = '($FORMAT_STRING_PLACEHOLDER)',
+    PERCENTAGE = 'FORMAT_STRING_PLACEHOLDER%',
+    SCIENTIFIC_NOTATION = 'FORMAT_STRING_PLACEHOLDERE+0'
 }
```

Comment:
Ok! These changes require you to touch a new part of the codebase -- the part that deals with upgrading analyses. 

# Why upgrading analyses

Imagine that a user uses Mito 0.1. and sets the format of some number column to `plain text`. This is saved in a file in the `.mito/saved_analyses/{analysis_id}.json` file on their computer. 

Then, they go and upgrade to Mito 0.2 -- the version that makes the above changes. That means that when the analysis is replayed, the number format that gets set is `plain text` -- but notably, the frontend no longer detects that as a valid number formatting, and so the number wouldn't be properly formatted.

**Whenever we change the structure or contents of a step performer, we have to do an upgrade, to make sure that previous user analyses don't break.** Notably, in this case, we need to upgrade the `set_dataframe_format.py` step!

There is a whole bunch of infrastructure for managing these upgrades (relevant files are `mitosheet/saved_analyses/step_upgraders/set_dataframe_format.py` and `mitosheet/saved_analyses/schema_utils.py` if you're interested). 

# Should we upgrade in this case

I'm wondering -- since we have to do transformations on the FORMAT_STRING_PLACEHOLDER anyways -- might we want to just keep the `NumberColumnFormatEnum` the same for now? 

You'll have to change the backend a bit (but not by a ton) just to have an If statement for each type of these - but otherwise the logic will be very similar to what you have. 

Then, if we want to move closer to Excel formatting strings over time, we can do it in the following way:
1. Add a new number format type: 'excel format string'. Notably, this will also take a format string as well an additional parameter on top of the precision. 
2. Furthermore, as this is a _new parameter_ and not an _old one_ -- we won't break anyone's old analyses. 
3. We can keep all the old code paths around for a while, until we think no one is using them anymore.

I am still excited about building a TSX function called `getFormattedNumber(num: number, excel_format_string: string) -> string` -- but I think that it might be a bigger lift than is worth it for now (especially with having to upgrade). So perhaps let's just make these changes as small as possible. 

To make it very clear, I think we should:
1. Change `NumberColumnFormatEnum` back to previous values.
2. Add if statements to the backend to build these strings.
3. Change the interface in the way you said (so you're passing the final Excel formatting strings to the add_formatting.. function)

Reply:
Got it! That all definitely makes sense. I'll make those changes!